mapstruct-2/src/main/java/com/baeldung/mapstruct/date/mapper/DateMapper.java

```diff
@@ -0,0 +1,18 @@
+package com.baeldung.mapstruct.date.mapper;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class DateMapper {
+
+    public Date mapStringToDate(String date) {
+        try {
+            SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd");
+            return dateFormat.parse(date);
+        } catch (ParseException e) {
```

Comment:
Is mapstruct unable to cope with checked exceptions? If it can cope with them, I'd prefer to see the checked exception declared than wrapped in a RuntimeException

As in: `public Date mapStringToDate(String date) throws ParseException {`